circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/RememberRetained.kt

```diff
@@ -144,7 +145,10 @@ public fun <T : Any> rememberRetained(
 ): T {
   val saveableStateRegistry = LocalSaveableStateRegistry.current
   val retainedStateRegistry = LocalRetainedStateRegistry.current
-
+  // Short-circuit no-ops
+  if (retainedStateRegistry === NoOpRetainedStateRegistry) {
```

Comment:
The likelihood of this is probably small, but would this do weird things if the value of `LocalRetainedStateRegistry.current` changed between being a `NoOpRetainedStateRegistry` and a functional one, or vice versa? I think that could result in lost state because the code paths would be entirely different?

Reply:
Not sure it'd be possible, `NoOpRetainedStateRegistry` is internal and is used as the default. So we'd be the ones jumping back and forth 😅  It would absolutely result in lost state if that did happen though.